messaging.rst

```diff
@@ -41,14 +35,23 @@ Fields
 +--------------------------+------------+--------------------------------------------------------------------------------+
 |  channel_identifier      | uint256    | Channel identifier inside the TokenNetwork contract                            |
 +--------------------------+------------+--------------------------------------------------------------------------------+
-|  message_hash            | bytes32    | Hash of the message                                                            |
-+--------------------------+------------+--------------------------------------------------------------------------------+
-|  signature               | bytes      | Elliptic Curve 256k1 signature on the above data                               |
```

Comment:
Why was the `message_hash` removed from the fields? Also, it's worth keeping the `additional_hash` naming consistent. An issue can be opened in the `raiden` repo to make this consistent in the implementation (if not done already).

Reply:
The implementation does not contain `message_hash` or `additional_hash` in the packed format.  For now, I'm following what @konradkonrad and @andrevmatos said in https://github.com/raiden-network/raiden/issues/3083 .  Once the discussion in https://github.com/raiden-network/raiden/issues/3083 settles, I can update this table.

Reply:
@pirapira what exactly does this table want to document? From the name I assume the packed [balance_proof](https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/messages.py#L311) that is sent in the wire as part of all EnvelopeMessages?

If so perhaps you can also mention that the actual balance proof contains a `balance_hash` which is the hash of `transferred_amount` + `locked_amount` + `locksroot`. So they are not really part of the balance proof but hashed inside.

Also the offichain balance proof contains a `msg_type` which is intended to be consumed by the contract and contains one of [these](https://github.com/raiden-network/raiden-contracts/blob/c6e94549c7296204b61742da174ef102961bbeb3/raiden_contracts/contracts/TokenNetwork.sol#L105) values.

Finally the [additional_hash](https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/transfer/balance_proof.py#L30) is still there.

But perhaps you wanted to document some other message?

Reply:
@LefterisJP the result of `pack_balance_proof()` is not contained in any messages.  It's just for calculating hashes.

Reply:
1. The spec says a `LockedTransfer` should contain an offchain balance proof.  https://github.com/raiden-network/spec/blob/4a2868c6728e7917d0847f68a0673135fdf10df0/messaging.rst#fields-3
2. The implementation says `LockedTransfer` does not contain an `additional_hash`.
https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/messages.py#L1102

These are why `additional_hash` is not part of an offchain balance proofl

Reply:
I meant it's contained as part of the signature in all the Balance proof containing messages since this is the data that is signed.

But still which message from [messages.py](https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/messages.py) is this table supposed to document then?

Reply:
This table documents `LockedTransfer - {lock, initiator, target}`.

Reply:
Hmm that does not seem right. I think what you are documenting is essentially the "Unlock" message, known in the codebase as [Secret](https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/messages.py#L645).

Cause if it was `LockedTransfer` you would also need a `token_address`, `lock_expiration`, `secret_hash`, `recipient` e.t.c. as seen [here](https://github.com/raiden-network/raiden/blob/dff9adfe0d9efe8d4dff46e457318cdbcc992125/raiden/messages.py#L1102)

Can you perhaps add a note that this is the message which you are trying to document for the implementation?

Reply:
> Maybe the whole thing is easier if we just document all fields of all real messages, without using this OffchainBalanceProof macro?

As discussed offline probably yes :)

Reply:
> Maybe the whole thing is easier if we just document all fields of all real messages, without using this OffchainBalanceProof macro?

Agree. We had this format in the beginning and I am still in favor of it, especially that it triggered so many discussions on naming.